lib/model/CDetectionRule.cc

```diff
@@ -54,6 +65,31 @@ bool CDetectionRule::apply(ERuleAction action,
     return true;
 }
 
+void CDetectionRule::executeCallback(CAnomalyDetectorModel& model, core_t::TTime time) const {
+
+    if (m_Callback) {
+        bool executeCallback{true};
+        for (const auto& condition : m_Conditions) {
+            executeCallback &= condition.test(time);
+        }
+        if (executeCallback) {
+            m_Callback(model);
+        }
+    }
+}
+
+void CDetectionRule::addTimeShift(core_t::TTime timeShift) {
+    this->setCallback([timeShift, this](CAnomalyDetectorModel& model) {
+        if (this->m_TimeShiftApplied == false) {
```

Comment:
I think this can be a mutable lambda and make `m_TimeShiftApplied` a member of it rather than the enclosing class. Seems cleaner. Also does the current approach work with multiple time shift rules?

Reply:
Multiple time shifts should live in the different detection rules, since every detection rule will get only on force_time_shift action. I'll add a unit test to verify the functionality 👍🏻

Reply:
A modified the lambda function to contain its own state. You are right, it's cleaner this way.